### PR TITLE
Andres/add rfq event listener

### PR DESCRIPTION
**Pull request by AndresElizondo**

### CODE REVIEW

File: src/events.ts [hunk 1]
```diff
@@ -232,6 +237,18 @@ function uniV3PoolSingletonCallback(pools: UniswapV3PoolCreatedEvent[]) {
     return pools;
 }
 
+function settlerContractSingletonCallback(settlerERC721TransferEvents: SettlerERC721TransferEvent[]) {
+    console.log('.');
```
Comment (Ktl-XV) [inline src/events.ts:241]: Should use logger from `./utils` instead of console.log

File: src/settlerContractSingleton.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            return;
+        }
+        logger.info('Loading Settler contract deployments to memory');
+        SettlerContractSingleton.instance = new SettlerContractSingleton();
+        const settlerContracts = await connection.query(
+            `
+            SELECT "to" AS address
+            FROM ${SCHEMA}.settler_erc721_transfer_events
+            WHERE "to" <> '0x0000000000000000000000000000000000000000'
+            ORDER BY block_number
```
Comment (Ktl-XV) [inline src/settlerContractSingleton.ts:28]: Order is not needed and removing it should make the query (imperceptibly) faster
Or use DESC to find recent contracts faster (probably no impact though)

File: src/parsers/events/rfq_order_event.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const totalSize = '0x'.length + orderHashSize + filledAmountSize;
+    if (eventLog.data.length !== totalSize) {
+        return null;
+    }
+
+    parseEvent(eventLog, rfqOrderEvent);
+
+    const settlerContractSingleton = SettlerContractSingleton.getInstance();
+    const settlerContracts = settlerContractSingleton.getContracts();
+    if (!settlerContracts.find((contract) => contract.address === rfqOrderEvent.contractAddress)) {
```
Comment (Ktl-XV) [inline src/parsers/events/rfq_order_event.ts:29]: Use `includes` instead of find?

File: src/settlerContractSingleton.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import { logger } from './utils';
+import { Connection } from 'typeorm';
+
+type SettlerContract = {
+    address: string;
+};
+
+export class SettlerContractSingleton {
+    private static instance: SettlerContractSingleton;
+    private contracts: SettlerContract[];
```
Comment (Ktl-XV) [inline src/settlerContractSingleton.ts:11]: Why not a simple string array?

**Review by Ktl-XV — APPROVED** [review]

File: src/settlerContractSingleton.ts (hunk 4 above)
Comment (AndresElizondo) [reply src/settlerContractSingleton.ts:11]: Used to have start and end block numbers for each contract deployment.
Will simplify it, thanks

**Pull request merged**